### PR TITLE
Upgrade crtm to 2.4.0 on WCOSS2 for GFS V16.3.0 Implementation

### LINKED ISSUE #549: Upgrade crtm to 2.4.0 for GFS V16.3.0 Implementation
To support GFS V16.3.0 implementation, an upgrade of crtm (2.4.0) is needed in GFS V16 branch release/gfs_v16.

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
Upgrade crtm to 2.4.0 to support GFS V16.3.0 implementation.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
Since crtm/2.4.0 is not available at the hpc-stack with intel/2018 for GFS V16 on Hera and Orion, updates for Hera and Orion will be incorporated in the future.

File: sorc/ncep_post.fd/build_upp_lib.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 SHELL=/bin/sh
 
-module purge
+module reset
```
Comment (aerorahul) [inline sorc/ncep_post.fd/build_upp_lib.sh]: I suggest remove this line altogether.
module reset will not work on Hera and Orion.
Besides, this being done in the if conditional blocks on line 34 and line 38.
```suggestion
```

**Review by aerorahul — COMMENTED** [review]
module and version changes look good.
I don't think I can comment on the Fortran code changes.

Just one suggestion for a change in `build_upp_lib.sh`, since it will break on Hera and Orion.

File: sorc/ncep_post.fd/build_upp_lib.sh (hunk 1 above)
Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/build_upp_lib.sh]: @aerorahul Updated. Thanks!

**Review by aerorahul — APPROVED** [review]
Looks good.

**Pull request merged**